### PR TITLE
Add benchmark for source-generated SetBinding

### PR DESCRIPTION
**Pull request by simonrozsival**
| Method                          | Mean     | Error    | StdDev   | Gen0   | Gen1   | Allocated |
|----------------------           |---------:|---------:|---------:|-------:|-------:|----------:|
| TypedBindName                   | 13.11 us | 0.244 us | 0.251 us | 1.2207 | 0.6104 |    7.5 KB |
| SourceGeneratedBindName         | 14.39 us | 0.137 us | 0.129 us | 1.3580 | 0.6714 |   8.36 KB |
| BindName                        | 16.98 us | 0.315 us | 0.279 us | 1.6785 | 0.8240 |  10.31 KB |
| TypedBindChild                  | 20.47 us | 0.214 us | 0.190 us | 2.0142 | 1.0071 |  12.42 KB |
| SourceGeneratedBindChild        | 21.09 us | 0.374 us | 0.349 us | 2.1667 | 1.0681 |  13.28 KB |
| BindChild                       | 21.51 us | 0.284 us | 0.266 us | 2.5940 | 1.2817 |  16.02 KB |
| TypedBindChildIndexer           | 21.78 us | 0.361 us | 0.338 us | 2.3193 | 1.1597 |  14.38 KB |
| SourceGeneratedBindChildIndexer | 22.46 us | 0.264 us | 0.247 us | 2.4719 | 1.2207 |  15.23 KB |
| BindChildIndexer                | 41.00 us | 0.459 us | 0.429 us | 4.1504 | 2.0752 |  25.55 KB |

### CODE REVIEW

File: src/Controls/src/BindingSourceGen/DiagnosticsFactory.cs [hunk 1]
```diff
@@ -56,7 +56,7 @@ public static DiagnosticInfo SuboptimalSetBindingOverload(Location location)
                 title: "SetBinding with string path",
                 messageFormat: "TODO: consider using SetBinding overload with a lambda getter",
                 category: "Usage",
-                defaultSeverity: DiagnosticSeverity.Warning,
-                isEnabledByDefault: true),
+                defaultSeverity: DiagnosticSeverity.Hidden,
```
[the remark this answers is not in the document]
Comment (simonrozsival) [reply src/Controls/src/BindingSourceGen/DiagnosticsFactory.cs]: Will this still be shown in the CLI output or in VS/VSCode?

File: src/Controls/src/BindingSourceGen/BindingSourceGenerator.cs [hunk 2]
```diff
@@ -106,15 +109,19 @@ static BindingDiagnosticsWrapper GetBindingForGeneration(GeneratorSyntaxContext
...
 		{
-			return new EquatableArray<DiagnosticInfo>([DiagnosticsFactory.SuboptimalSetBindingOverload(getter.GetLocation())]);
+			var type = context.SemanticModel.GetTypeInfo(secondArgument, cancellationToken: t).Type;
+			if (type != null && type.Name == "Func")
+			{
+				return new EquatableArray<DiagnosticInfo>([DiagnosticsFactory.GetterIsNotLambda(secondArgument.GetLocation())]);
+			}
+			else // String and Binding
+			{
+				return new EquatableArray<DiagnosticInfo>([DiagnosticsFactory.SuboptimalSetBindingOverload(secondArgument.GetLocation())]);
```
Comment (simonrozsival) [inline src/Controls/src/BindingSourceGen/BindingSourceGenerator.cs:123]: Does this include `SetBinding(P, MyMethod);`?

Comment (simonrozsival) [reply src/Controls/src/BindingSourceGen/BindingSourceGenerator.cs:123]: Could you please add a test case that covers it?

File: src/Controls/src/BindingSourceGen/DiagnosticsFactory.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (jkurdek) [reply src/Controls/src/BindingSourceGen/DiagnosticsFactory.cs]: I tried running the sample app with different overload, did not see any warnings/errors. When passing func it shows correct error.

**Pull request merged**